sos/policies/distros/__init__.py

```diff
@@ -762,6 +917,71 @@ def upload_ftp(self, url=None, directory=None, user=None, password=None):
         except IOError:
             raise Exception("could not open archive file")
 
+    def upload_s3(self, endpoint=None, region=None, bucket=None, prefix=None,
+                  access_key=None, secret_key=None):
+        """Attempts to upload the archive to an S3 bucket.
+
+        :param endpoint: The S3 endpoint to upload to
+        :type endpoint: str
+
+        :param region: The S3 region to upload to
+        :type region: str
+
+        :param bucket: The name of the S3 bucket to upload to
+        :type bucket: str
+
+        :param prefix: The prefix for the S3 object/key
+        :type prefix: str
+
+        :param access_key: The access key for the S3 bucket
+        :type access_key: str
+
+        :param secret_key: The secret key for the S3 bucket
+        :type secret_key: str
+
+        :returns: True if upload is successful
+        :rtype: bool
+
+        :raises: Exception if upload is unsuccessful
+        """
+        if not BOTO3_LOADED:
+            raise Exception("Unable to upload due to missing python boto3 "
+                            "library")
```

Comment:
Ack.

Reply:
I presume this is intended as "acknowledge" for acceptance of a requested/suggested change.